modules/stochastic_tools/src/surrogates/ExponentialCovariance.C

```diff
@@ -0,0 +1,92 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ExponentialCovariance.h"
+#include <cmath>
+
+registerMooseObject("StochasticToolsApp", ExponentialCovariance);
+
+InputParameters
+ExponentialCovariance::validParams()
+{
+  InputParameters params = CovarianceFunctionBase::validParams();
+  params.addClassDescription("Exponential covariance function.");
+  params.addRequiredParam<std::vector<Real>>("length_factor",
+                                             "Length Factor to use for Covariance Kernel");
+  params.addRequiredParam<Real>("signal_variance",
+                                "Signal Variance (sigma_f^2) to use for kernel calculation.");
+  params.addRequiredParam<Real>("noise_variance",
+                                "Noise Variance (sigma_n^2) to use for kernel calculation.");
+  params.addRequiredParam<Real>("gamma", "Gamma to use for Exponential Covariance Kernel");
+  return params;
+}
+
+ExponentialCovariance::ExponentialCovariance(const InputParameters & parameters)
+  : CovarianceFunctionBase(parameters),
+    _length_factor(getParam<std::vector<Real>>("length_factor")),
+    _sigma_f_squared(getParam<Real>("signal_variance")),
+    _sigma_n_squared(getParam<Real>("noise_variance")),
+    _gamma(getParam<Real>("gamma"))
+{
+}
+
+void
+ExponentialCovariance::buildHyperParamMap(
+    std::unordered_map<std::string, Real> & map,
+    std::unordered_map<std::string, std::vector<Real>> & vec_map) const
+{
+  map["signal_variance"] = _sigma_f_squared;
+  map["noise_variance"] = _sigma_n_squared;
+  map["gamma"] = _gamma;
+
+  vec_map["length_factor"] = _length_factor;
+}
+
+void
+ExponentialCovariance::computeCovarianceMatrix(RealEigenMatrix & K,
+                                               const RealEigenMatrix & x,
+                                               const RealEigenMatrix & xp,
+                                               const bool is_self_covariance) const
+{
+  ExponentialFunction(
```

Comment:
Is there a value in creating a separate exponential function for this? Why is it not enough to write a specific computeCovarianceMatrix function in all of the Cov classes?

Reply:
Yes, the static function allows for unit tests to be employed, which will be implemented in a future MR. Ian was having trouble with the unit tests, so I will probably be doing this.

Reply:
This is a recent change to support unit testing. Initially it was separate computeCovariacneMatrix() for each. This new way allows for the actual math to be implemented as a static function, which is helpful for testing.